desktop/src/main.rs

```diff
@@ -539,21 +560,38 @@ impl App {
                         let movie_width = swf_header.stage_size().width().to_pixels();
                         let movie_height = swf_header.stage_size().height().to_pixels();
 
+                        let monitors = Window::available_monitors(&self.window);
+                        let max_screen_size = get_max_screen_size(monitors);
+
                         let window_size: Size = match (self.opt.width, self.opt.height) {
-                            (None, None) => LogicalSize::new(movie_width, movie_height).into(),
+                            (None, None) => LogicalSize::new(
+                                movie_width.clamp(16.0, max_screen_size.0 as f64),
```

Comment:
To avoid repeating the same `clamp`s 4 times, winit's [`Size::clamp`](https://docs.rs/winit/latest/winit/dpi/enum.Size.html#method.clamp) can be used.
Also I think `max_inner_size` (and `min_inner_size`) should be stored in `App` itself so it can be retrieved here rather than computed again.